src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala

```diff
@@ -42,4 +51,17 @@ trait RestJsonClient {
       }
     }
   }
+
+  def unAuthedRequestToObject[T](req: HttpRequest, compressed: Boolean = false)(implicit unmarshaller: Unmarshaller[T]): Future[T] = {
+    unAuthedRequest(req, compressed) map { response =>
+      response.status match {
+        case s if s.isSuccess =>
+          response.entity.as[T] match {
+            case Right(obj) => obj
+            case Left(error) => throw new FireCloudExceptionWithErrorReport(FCErrorReport(response))
+          }
+        case f => throw new FireCloudExceptionWithErrorReport(FCErrorReport(response))
+      }
+    }
+  }
```

Comment:
also here, maybe something like:
```
  def requestToObject[T](req: HttpRequest, compressed: Boolean = false)(implicit userInfo: WithAccessToken, unmarshaller: Unmarshaller[T]): Future[T] = {
    doRequest( true, req, compressed )
  }

  def unAuthedRequestToObject[T](req: HttpRequest, compressed: Boolean = false)(implicit unmarshaller: Unmarshaller[T]): Future[T] = {
    doRequest( false, req, compressed )
  }

  private def doRequest[T](auth: Boolean, req: HttpRequest, compressed: Boolean = false)(implicit unmarshaller: Unmarshaller[T]): Future[T] = {
    val resp = if (auth)
      userAuthedRequest( req, compressed )
    else
      unAuthedRequest(req, compressed)
    
    resp map { response =>
      response.status match {
        case s if s.isSuccess =>
          response.entity.as[T] match {
            case Right(obj) => obj
            case Left(error) => throw new FireCloudExceptionWithErrorReport(FCErrorReport(response))
          }
        case f => throw new FireCloudExceptionWithErrorReport(FCErrorReport(response))
      }
    }
  }

```
( might want to refactor the two public methods to `authedRequestToObject` and `unAuthedRequestToObject` for clarity )

I haven't actually tested this, or the previous comment, but you get the idea!